custom_components/tahoma/light.py

```diff
@@ -131,26 +134,29 @@ def effect(self) -> str:
 
     def update(self):
         """Fetch new state data for this light."""
-
         if self.should_wait():
             self.schedule_update_ha_state(True)
             return
 
         self.controller.get_states([self.tahoma_device])
 
-        if "core:LightIntensityState" in self.tahoma_device.active_states:
-            self._brightness = self.tahoma_device.active_states.get(
-                "core:LightIntensityState"
-            )
+        states = self.tahoma_device.active_states
+
+        if CORE_LIGHT_INTENSITY_STATE in states:
+            self._brightness = states.get(CORE_LIGHT_INTENSITY_STATE)
 
-        if self.tahoma_device.active_states.get("core:OnOffState") == "on":
+        if states.get(CORE_ON_OFF_STATE) == "on":
```

Comment:
```suggestion
self._state = states.get(CORE_ON_OFF_STATE) == "on":
```

Reply:
My remark was to replace the if AND else block.

Reply:
@tetienne I understood, GitHub doesn't support suggestions on multiple lines yet... I have fixed it manually, good suggestion.